packages/store-ui/src/atoms/List/List.tsx

```diff
@@ -0,0 +1,56 @@
+import type { HTMLAttributes, ElementType } from 'react'
+import React, { forwardRef } from 'react'
+
+export type ListVariant = 'default' | 'description' | 'ordered' | 'unordered'
+
+const variantToType = (variant: ListVariant) => {
+  switch (variant) {
+    case 'description':
+      return 'dl'
+
+    case 'unordered':
+      return 'ul'
+
+    case 'ordered':
+      return 'ol'
+
+    default:
+      return 'div'
+  }
+}
+
+export interface ListProps extends HTMLAttributes<HTMLElement> {
+  /**
+   * Set the HTML element tag of this component.
+   */
+  as?: ElementType
+  /**
+   * ID to find this component in testing tools (e.g.: cypress, testing library, and jest).
+   */
+  testId?: string
+  /**
+   * The current use case variants for Lists.
+   */
+  variant?: ListVariant
+}
+
+export const List = forwardRef<HTMLElement, ListProps>(function List(
```

Comment:
Remember to check the type of this component, because the polymorphic behavior may cause some problems during the build when using typescript.